### ISSUE TITLE
Code Signing Error with Xcode 15+

### ISSUE THREAD

**Issue by JawadJaber** [dotnet/maui#18231]
### Description

I have upgraded my MAC to Mac OS Sonomo, and Upgraded Xcode to version 15.0.1. After that I tried to publish my app to app store but failed with the following message. Pls note that I tried with both .NET7 and .NET8 latest versions.

I have tried to re-issue the certificates in Xcode and Apple platform but all have same result.

and also tried to create new .NET MAUI app with same results.

`C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.8968-net8-rc2\tools\msbuild\iOS\Xamarin.Shared.targets(2802,3): error : Code signing must be enabled to create an Xcode archive.`

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

7.0.92

### Affected platforms

iOS

### Affected platform versions

iOS 15

### Did you find any workaround?

No

**Comment by jfversluis** [comment]
@rolfbjarne I think this lives more on your level, do you have any ideas here?

@JawadJaber could you maybe share more about _how_ you try to publish your app? Is it through the command-line? What is the command you are using? Is it through VS, what are the steps that you take? Thanks!

**Comment by rolfbjarne** [comment]
> @rolfbjarne I think this lives more on your level, do you have any ideas here?

There's not information here to even guess, in addition to the questions you've asked it might also be helpful to get [binlogs](https://aka.ms/binlog)

**Comment by JawadJaber** [comment]
> @rolfbjarne I think this lives more on your level, do you have any ideas here?
> 
> @JawadJaber could you maybe share more about _how_ you try to publish your app? Is it through the command-line? What is the command you are using? Is it through VS, what are the steps that you take? Thanks!

It is comman-line, as follows:
dotnet publish C:\ProofMauiProject\ProofMaui\ProofMauiProject\MauiApp1\Maui.Mobile.Dev.csproj -c:Release -f:net8.0-ios /p:ArchiveOnBuild=true /p:RuntimeIdentifier=ios-arm64 /p:ServerAddress=192.168.100.255 /p:ServerUser=jawadsaber /p:TcpPort=58181 /p:ServerPassword=**** /p:EnableAssemblyILStripping=false /p:_DotNetRootRemoteDirectory=/Users/jawadsaber/Library/Caches/Xamarin/XMA/SDKs/dotnet/

Pls note this is not the first time to publish the app on the App store, and I always follow the steps you mentioned in this youtube video
https://www.youtube.com/watch?v=kpZi5xAvpZA

Moreover, I have reissued all certificates and provisioning profiles with same output.

**Comment by JawadJaber** [comment]
I would like to close this case as it is resolved after two days of continuous effort, I am not sure what was wrong but I did the following steps:
1. I was using iPhone Distribution certificate while I have to use Apple Distribution as Xcode logged that it failed to create the ipa file at different platforms. May be because I am using .NET MAUI class library.
2. I have followed the following SO case [https://stackoverflow.com/questions/69049965/failed-to-codesign-libskiasharp](url)
3. May be because the cproj file contained two signing information one for distribution and one or development, the development part was not specifically mentioned in Debug mode and it was after the release section as follows

``
